lib/util/rule-codes.js

```diff
@@ -2,23 +2,25 @@
 
 module.exports = ruleCodes
 
+var NO_CODES = []
+
 // Parse rule codes.
 function ruleCodes(flags, value) {
   var flag = flags.FLAG
-  var result = []
+  var result
   var length
   var index
 
-  if (!value) {
-    return result
-  }
+  if (!value) return NO_CODES
 
   if (flag === 'long') {
     index = 0
     length = value.length
 
+    result = new Array(Math.ceil(length / 2))
```

Comment:
```suggestion
    result = []
```

Is the current needed? Or can it be simpler?

Reply:
Those two lines are logically equivalent in the context of how that function works, i.e. they will always lead to the same result, but using `new Array` preallocates an array of the right size immediately, while growing a 0-length one will be more expensive, both because its size will need to be changed at runtime and because v8 may overshoot it and think that it will actually grow to contain 10 items (maybe because it went from 0-length to 6-length quickly) while in reality it will stop at 6 or something.

Reply:
If it is a significant improvement, can you add a comment above that line stating that it shaved off x ms or so?

Reply:
According to my understanding of how stuff like v8 work that's an improvement for sure, I can add a comment about that but attaching a precise number of ms that that shaved off would be challenging, there are too many factors that go into each benchmark execution.

Reply:
First 3 runs with `[]`, remaining 3 runs with `new Array`:

![image](https://user-images.githubusercontent.com/1812093/93786618-6ebc9c00-fc27-11ea-804d-3eccc14e9699.png)

I'm not sure how much one should see into this though, I should be running benchmarks like a million times to get something statistically significant proabably.